typed-racket-lib/typed-racket/base-env/prims-struct.rkt

```diff
@@ -121,38 +126,48 @@
 
 
 ;; User-facing macros for defining typed structure types
-(define-syntaxes (define-typed-struct -struct)
-  (values
-   (lambda (stx)
-     (syntax-parse stx
-       [(_ vars:maybe-type-vars nm:struct-name (fs:fld-spec ...)
-           opts:struct-options)
-        (let ([mutable? (if (attribute opts.mutable?) #'(#:mutable) #'())]
-              [cname (second (build-struct-names #'nm.name null #t #t))]
-              [prefab? (if (attribute opts.prefab?) #'(#:prefab) #'())])
-          (with-syntax ([d-s (ignore-some
-                               (syntax/loc stx (define-struct nm (fs.fld ...) . opts)))]
-                        [dtsi (quasisyntax/loc stx
-                                (dtsi* (vars.vars ...) nm (fs.form ...)
-                                       #:maker #,cname
-                                       #,@mutable?
-                                       #,@prefab?))])
-            #'(begin d-s dtsi)))]))
-   (lambda (stx)
-     (syntax-parse stx
-       [(_ vars:maybe-type-vars nm:struct-name/new (fs:fld-spec ...)
-           opts:struct-options)
-        (let ([mutable? (if (attribute opts.mutable?) #'(#:mutable) #'())]
-              [prefab? (if (attribute opts.prefab?) #'(#:prefab) #'())])
-          (with-syntax ([d-s (ignore (quasisyntax/loc stx
-                                       (struct #,@(attribute nm.new-spec) (fs.fld ...)
-                                               . opts)))]
-                        [dtsi (quasisyntax/loc stx
-                                (dtsi* (vars.vars ...)
-                                       nm.old-spec (fs.form ...)
-                                       #,@mutable?
-                                       #,@prefab?))])
-            #'(begin d-s dtsi)))]))))
+(define-syntax (define-typed-struct stx)
+  (syntax-parse stx
+    [(_ vars:maybe-type-vars nm:struct-name fs opts:struct-options)
```

Comment:
The other three are checked only because they are splicing syntax classes, so they have to be there. I could also check the fields, but `-struct` will also do that.